jupyterhub_moss/utils.py

```diff
@@ -44,6 +44,26 @@ def parse_timelimit(timelimit: str) -> datetime.timedelta:
     )
 
 
+_GPU_GRES_REGEXP = re.compile(r"^gpu:(?P<type_>[^:]+):(?P<count>[0-9]+)(?:\(.*\))?$")
+
+
+def parse_gpu_resource(generic_resources: str) -> tuple[str, str]:
+    """Parse SLURM generic resources (aka., gres) to retrieve GPU information.
+
+    Information about the first GPU found is returned if any.
+
+    :returns: (GPU resource template, number of GPUs) or None if the resource is not a GPU.
+    :raises ValueError: If parsing failed.
+    """
+    for resource in generic_resources.split(","):
```

Comment:
SLURM "generic resources" can be a comma-separated list of many different things, here we only pick the first `gpu` with a correct syntax.